extensions-support/azure-core-http-client-vertx/deployment/src/test/java/org/apache/camel/quarkus/support/azure/core/http/vertx/VertxHttpClientBuilderTests.java

```diff
@@ -0,0 +1,164 @@
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.support.azure.core.http.vertx;
+
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.util.Configuration;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClientOptions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import reactor.test.StepVerifier;
+
+import static org.apache.camel.quarkus.support.azure.core.http.vertx.VertxHttpClientTestResource.PROXY_PASSWORD;
+import static org.apache.camel.quarkus.support.azure.core.http.vertx.VertxHttpClientTestResource.PROXY_USER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests {@link VertxHttpClientBuilder}.
+ */
+public class VertxHttpClientBuilderTests {
+    private static final String COOKIE_VALIDATOR_PATH = "/cookieValidator";
+    private static final String DEFAULT_PATH = "/default";
+    private static final String DISPATCHER_PATH = "/dispatcher";
+
+    private static final WireMockServer server = new WireMockServer(
+            WireMockConfiguration.options().dynamicPort().disableRequestJournal());
+    private static final Vertx vertx = Vertx.vertx();
+
+    private static String defaultUrl;
+
+    @BeforeAll
+    public static void setupWireMock() {
+        // Mocked endpoint to test building a client with a prebuilt client.
+        server.stubFor(WireMock.get(COOKIE_VALIDATOR_PATH).withCookie("test", WireMock.matching("success"))
+                .willReturn(WireMock.aResponse().withStatus(200)));
+
+        // Mocked endpoint to test building a client with a timeout.
+        server.stubFor(WireMock.get(DEFAULT_PATH).willReturn(WireMock.aResponse().withStatus(200)));
+
+        // Mocked endpoint to test building a client with a dispatcher and uses a delayed response.
+        server.stubFor(WireMock.get(DISPATCHER_PATH).willReturn(WireMock.aResponse().withStatus(200)
+                .withFixedDelay(5000)));
+
+        server.start();
+
+        defaultUrl = "http://localhost:" + server.port() + DEFAULT_PATH;
+    }
+
+    @AfterAll
+    public static void shutdownWireMock() {
+        if (server.isRunning()) {
+            server.shutdown();
+        }
+        vertx.close();
+    }
+
+    @Test
+    public void buildWithConfigurationNone() {
+        HttpClient client = new VertxHttpClientBuilder(vertx)
```

Comment:
It might be beneficial to close the client.
It will be closed automatically when vert.x is closed, but you only close vert.x after all the tests.